mhs/outbound/outbound/request/synchronous/handler.py

```diff
@@ -179,12 +179,13 @@ def _parse_body(self):
                                         reason=f'Invalid request. Validation errors: {validation_errors}') from e
         return parsed_body.payload
 
-    def _extract_sync_async_header(self):
-        sync_async_header = self.request.headers.get(HttpHeaders.SYNC_ASYNC, None)
-        if not sync_async_header:
-            logger.error('Failed to parse sync-async header from message')
-            raise tornado.web.HTTPError(400, 'Sync-Async header missing', reason='Sync-Async header missing')
-        if sync_async_header == 'true':
+    def _extract_wait_for_response_header(self):
+        wait_for_response_header = self.request.headers.get(HttpHeaders.WAIT_FOR_RESPONSE, None)
+        if not wait_for_response_header:
+            logger.error('Failed to parse wait-for-response header from message')
+            raise tornado.web.HTTPError(400, 'wait-for-response header missing',
+                                        reason='wait-for-response header missing')
+        if wait_for_response_header.lower() == 'true':
             return True
         else:
             return False
```

Comment:
(Optional) This implementation assumes anything different than `True/true` as `False` (so i.e. `AAA` is also false). But all falsy values (like `0`) will show up as header is missing. Do we keep it that way?

Reply:
@petervdm, should the case of any string other than true or false be taken as false or should the exception by thrown?

Reply:
Well spotted @grzegorz-robak  The test partner reported this yesterday and I was going to raise another ticket for it, but you might as well do it in here @Alicja-Kainos

Reply:
@petervdm , I'll do it here :) But I need to know if aaa should be treated like an exception or like false. I think this should be an exception, shouldn't be?

Reply:
Yes an exception